### PR TITLE
fix/ci small things

### PR DESCRIPTION
**Pull request by altergui**
- test: allow setting chainID and height, useful to test soft-forks
- test: AdvanceTestBlocksUntilHeight method
- deepsource: enable test-coverage and other analyzers

### CODE REVIEW

File: vochain/apptest.go [hunk 1]
```diff
@@ -115,8 +123,37 @@ func (app *BaseApplication) AdvanceTestBlock() {
...
-	time.Sleep(time.Millisecond * 50)
+	time.Sleep(time.Microsecond * 50)
 	nextStartTime := time.Now()
 	app.testMockBlockStore.NewBlock(newHeight)
 	app.beginBlock(nextStartTime, uint32(newHeight))
 }
+
+// AdvanceTestBlocksUntilHeight is equivalent to AdvanceTestBlock,
+// but loops until reaching height n
+func (app *BaseApplication) AdvanceTestBlocksUntilHeight(n uint32) {
```
Comment (p4u) [inline vochain/apptest.go:134]: why this function does not use AdvanceTestBlock ?

Comment (altergui) [reply vochain/apptest.go:134]: i was unsure about reducing time.Sleep of AdvanceTestBlock would break things. fixed.

**Comment by altergui** [comment]
TestAddVote sometimes hits a race, maaaaybe because of reducing time.Sleep in AdvanceTestBlock. i'll debug this further.

```
==================
WARNING: DATA RACE
Write at 0x00c0007bc038 by goroutine 658:
  github.com/cockroachdb/pebble.newIndexedBatch()
      /home/runner/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20230620232302-06034ff014e0/batch.go:371 +0x1a4
  github.com/cockroachdb/pebble.(*DB).NewIndexedBatch()
      /home/runner/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20230620232302-06034ff014e0/db.go:1424 +0xc4
  go.vocdoni.io/dvote/db/pebbledb.(*PebbleDB).WriteTx()
      /home/runner/work/vocdoni-node/vocdoni-node/db/pebbledb/pebledb.go:138 +0x6a
  go.vocdoni.io/dvote/statedb.(*StateDB).BeginTx()
      /home/runner/work/vocdoni-node/vocdoni-node/statedb/statedb.go:301 +0x10f
  go.vocdoni.io/dvote/vochain/state.initStateDB()
      /home/runner/work/vocdoni-node/vocdoni-node/vochain/state/state.go:168 +0x451
  go.vocdoni.io/dvote/vochain/state.NewState()
      /home/runner/work/vocdoni-node/vocdoni-node/vochain/state/state.go:106 +0x14a
  go.vocdoni.io/dvote/test/testcommon.NewVochainState()
      /home/runner/work/vocdoni-node/vocdoni-node/test/testcommon/vochain.go:86 +0x94
  go.vocdoni.io/dvote/test/testcommon.NewVochainStateWithProcess()
      /home/runner/work/vocdoni-node/vocdoni-node/test/testcommon/vochain.go:129 +0x79
  go.vocdoni.io/dvote/test.TestAddVote()
      /home/runner/work/vocdoni-node/vocdoni-node/test/statedb_test.go:93 +0x44
  testing.tRunner()
      /opt/hostedtoolcache/go/1.21.1/x64/src/testing/testing.go:1595 +0x238
  testing.(*T).Run.func1()
      /opt/hostedtoolcache/go/1.21.1/x64/src/testing/testing.go:1648 +0x44
Previous read at 0x00c0007bc038 by goroutine 661:
  github.com/cockroachdb/pebble.(*Batch).release()
      /home/runner/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20230620232302-06034ff014e0/batch.go:388 +0x3c
  github.com/cockroachdb/pebble.(*Batch).Close()
      /home/runner/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20230620232302-06034ff014e0/batch.go:1283 +0x70
  go.vocdoni.io/dvote/db/pebbledb.WriteTx.Discard()
      /home/runner/work/vocdoni-node/vocdoni-node/db/pebbledb/pebledb.go:96 +0xe
  go.vocdoni.io/dvote/db/prefixeddb.(*PrefixedWriteTx).Discard()
      /home/runner/work/vocdoni-node/vocdoni-node/db/prefixeddb/prefixeddb.go:129 +0x9c
  go.vocdoni.io/dvote/vochain/state.(*State).Close()
      /home/runner/work/vocdoni-node/vocdoni-node/vochain/state/state.go:470 +0x183
  go.vocdoni.io/dvote/test/testcommon.NewVochainState.func1()
      /home/runner/work/vocdoni-node/vocdoni-node/test/testcommon/vochain.go:90 +0x44
  testing.(*common).Cleanup.func1()
      /opt/hostedtoolcache/go/1.21.1/x64/src/testing/testing.go:1169 +0x182
  testing.(*common).runCleanup()
      /opt/hostedtoolcache/go/1.21.1/x64/src/testing/testing.go:1347 +0x1c2
  testing.tRunner.func2()
      /opt/hostedtoolcache/go/1.21.1/x64/src/testing/testing.go:1589 +0x50
  runtime.deferreturn()
      /opt/hostedtoolcache/go/1.21.1/x64/src/runtime/panic.go:477 +0x30
  testing.(*T).Run.func1()
      /opt/hostedtoolcache/go/1.21.1/x64/src/testing/testing.go:1648 +0x44
Goroutine 658 (running) created at:
  testing.(*T).Run()
      /opt/hostedtoolcache/go/1.21.1/x64/src/testing/testing.go:1648 +0x82a
  testing.runTests.func1()
      /opt/hostedtoolcache/go/1.21.1/x64/src/testing/testing.go:2054 +0x84
  testing.tRunner()
      /opt/hostedtoolcache/go/1.21.1/x64/src/testing/testing.go:1595 +0x238
  testing.runTests()
      /opt/hostedtoolcache/go/1.21.1/x64/src/testing/testing.go:2052 +0x896
  testing.(*M).Run()
      /opt/hostedtoolcache/go/1.21.1/x64/src/testing/testing.go:1925 +0xb57
  main.main()
      _testmain.go:105 +0x2e4
Goroutine 661 (finished) created at:
  testing.(*T).Run()
      /opt/hostedtoolcache/go/1.21.1/x64/src/testing/testing.go:1648 +0x82a
  testing.runTests.func1()
      /opt/hostedtoolcache/go/1.21.1/x64/src/testing/testing.go:2054 +0x84
  testing.tRunner()
      /opt/hostedtoolcache/go/1.21.1/x64/src/testing/testing.go:1595 +0x238
  testing.runTests()
      /opt/hostedtoolcache/go/1.21.1/x64/src/testing/testing.go:2052 +0x896
  testing.(*M).Run()
      /opt/hostedtoolcache/go/1.21.1/x64/src/testing/testing.go:1925 +0xb57
  main.main()
      _testmain.go:105 +0x2e4
==================
--- FAIL: TestAddVote (0.02s)
    testing.go:1465: race detected during execution of test
```

**Comment by altergui** [comment]
...or maybe i just reproduced https://github.com/vocdoni/vocdoni-node/issues/1139 which would make me happy

**Comment by altergui** [comment]
the CI failure is definitely unrelated, so this PR is ready to approve/merge

**Pull request merged**